plugins/pulp_rpm/plugins/importers/yum/repomd/metadata.py

```diff
@@ -301,7 +318,13 @@ def generate_dbs(self):
                         db_key = self.generate_db_key(unit_key)
                         db_file_handle[db_key] = raw_xml
                     db_file_handle.sync()
+                    package_count[filename] = len(db_file_handle)
```

Comment:
FWIW, I probably would have approached this without the ``package_count`` dictionary. You could consider this here:

```
if self.rpm_count is None:
  self.rpm_count = len(db_file_handle)
elif self.rpm_count != len(db_file_handle):
  raise PulpCodedException(...)
```

See if you like that any better. If not, feel free to keep what you have.